cpp/arcticdb/version/version_core.cpp

```diff
@@ -789,22 +789,28 @@ void read_incompletes_to_pipeline(
     if(incomplete_segments.empty())
         return;
 
+    const auto* seg = &incomplete_segments.front().segment(store);
```

Comment:
Why is this now a pointer? What was it before?

Reply:
It was a reference pointing to the first segment in the list. However the first segment can be empty and then the normalization metadata is wrong (this is what caused the problem with wrong column names). Now I'm scanning until I find the first non-empty segment. It's easier to keep track using a pointer.

Reply:
Ah I get it. What's the behaviour if all of the segment row counts are 0? Line 792 means we'll still have `seg` point at the first incomplete segment, which doesn't seem intentional. If it was initialised to `nullptr`, and then was still `nullptr` after the loop below, what should we do?

Reply:
This shouldn't be nullptr in any case. It's initialized with the first element of incomplete_segments. If incomplete segments is empty the function returns (one line above ln. 790). If all segments are empty it will get the default normalization metadata an empty dataframe has.

Reply:
OK, this logic is extremely non-obvious, please add some comments to explain why it works